stdlib/ballerina-http/src/main/ballerina/http/http_secure_client.bal

```diff
@@ -47,12 +48,15 @@ public type HttpSecureClient object {
     @Return {value:"Error occured during HTTP client invocation"}
     public function post(string path, Request? request = ()) returns (Response|HttpConnectorError) {
         Request req = request ?: new;
-        var details = prepareRequest(req, config);
-        match details {
-            () => {}
-            HttpConnectorError err => return err;
+        check generateSecureRequest(req, config);
+        Request newOutRequest = check cloneRequest(req);
```

Comment:
Can't we clone only we have refresh token or only if we can retry?

Reply:
Yes. We can do that. I will fix that and update the PR.

Reply:
Fixed.